src/main/java/bio/terra/service/profile/ProfileApiController.java

```diff
@@ -118,38 +119,38 @@ protected void initBinder(final WebDataBinder binder) {
     }
 
     @Override
-    public ResponseEntity<BillingProfileModel> retrieveProfile(String id) {
+    public ResponseEntity<BillingProfileModel> retrieveProfile(UUID id) {
         AuthenticatedUserRequest user = authenticatedUserRequestFactory.from(request);
-        BillingProfileModel profileModel = profileService.getProfileById(id, user);
+        BillingProfileModel profileModel = profileService.getProfileById(id.toString(), user);
         return new ResponseEntity<>(profileModel, HttpStatus.OK);
     }
 
     @Override
     public ResponseEntity<PolicyResponse> addProfilePolicyMember(
-        @PathVariable("id") String id,
+        @PathVariable("id") UUID id,
         @PathVariable("policyName") String policyName,
         @Valid @RequestBody PolicyMemberRequest policyMember) {
         AuthenticatedUserRequest user = authenticatedUserRequestFactory.from(request);
-        PolicyModel policy = profileService.addProfilePolicyMember(id, policyName, policyMember, user);
+        PolicyModel policy = profileService.addProfilePolicyMember(id.toString(), policyName, policyMember, user);
         PolicyResponse response = new PolicyResponse().policies(Collections.singletonList(policy));
         return new ResponseEntity<>(response, HttpStatus.OK);
     }
 
     @Override
     public ResponseEntity<PolicyResponse> deleteProfilePolicyMember(
-        @PathVariable("id") String id,
+        @PathVariable("id") UUID id,
         @PathVariable("policyName") String policyName,
         @PathVariable("memberEmail") String memberEmail) {
         AuthenticatedUserRequest user = authenticatedUserRequestFactory.from(request);
-        PolicyModel policy = profileService.deleteProfilePolicyMember(id, policyName, memberEmail, user);
+        PolicyModel policy = profileService.deleteProfilePolicyMember(id.toString(), policyName, memberEmail, user);
         PolicyResponse response = new PolicyResponse().policies(Collections.singletonList(policy));
         return new ResponseEntity<>(response, HttpStatus.OK);
     }
 
     @Override
-    public ResponseEntity<PolicyResponse> retrieveProfilePolicies(@PathVariable("id") String id) {
+    public ResponseEntity<PolicyResponse> retrieveProfilePolicies(@PathVariable("id") UUID id) {
         AuthenticatedUserRequest user = authenticatedUserRequestFactory.from(request);
-        List<PolicyModel> policies = profileService.retrieveProfilePolicies(id, user);
+        List<PolicyModel> policies = profileService.retrieveProfilePolicies(id.toString(), user);
```

Comment:
You can change `ProfileService. retrieveProfilePolicies()` to accept `UUID` as this is the only caller and it end up calling `UUID.fromString()` anyway.